techhive-ui/src/components/NewInternship/NewInternship.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useContext, useState } from 'react';
+import axios from 'axios';
+import { RefreshContext } from '../Internships/Internships.jsx'; 
+import { useNavigate } from 'react-router-dom';
+import './NewInternship.css';
+import { IoIosCloseCircleOutline } from 'react-icons/io';
+
+const NewInternship = () => {
+  const [internshipData, setInternshipData] = useState({
+    title: '',
+    link: '',
+    description: '',
+    company: '',
+    category: '',
+    picture: null,
+  });
+  const [errors, setErrors] = useState([]);
+  const setRefreshData = useContext(RefreshContext);
+  const navigate = useNavigate();
+
+  const handleInputChange = (e) => {
+    const inputValue = e.target.value.trim();
+    
+    if (e.target.name === 'category') {
+      const categories = inputValue.split(',').map(item => item.trim());
+      setInternshipData({ ...internshipData, [e.target.name]: JSON.stringify(categories) });
+    } else {
+      setInternshipData({ ...internshipData, [e.target.name]: inputValue });
+    }
+}
+  const handleFileChange = (e) => {
+    setInternshipData({ ...internshipData, picture: e.target.files[0] });
+  }
+
+  const handleFormSubmit = async (e) => {
+    e.preventDefault();
+
+    const formData = new FormData();
+    Object.entries(internshipData).forEach(([key, value]) => formData.append(key, value));
+    console.log(internshipData);
```

Comment:
can remove

Reply:
Okay I will, sorry about the console logs. I will leave a video in workplace on how the internshipsData works.